inc/listeners/heartbeat.h

```diff
@@ -12,6 +12,12 @@
 
 class Heartbeat {
 public:
+  std::ostream* std_out;
+
+  Heartbeat(std::ostream* so) {

```

Comment:
```suggestion
  explicit Heartbeat(std::ostream* so) {

```

Danger: implicit conversions ahead.